tests/cpp/test_multidevice_tutorial.cpp

```diff
@@ -615,4 +615,509 @@ TEST_F(MultiDeviceTutorial, TensorShardingAndResharding) {
+  at::Tensor aten_tvb = at::randn({K, N}, options);
+  at::Tensor aten_tvd = at::empty({M / communicator_->size(), N}, options);
+
+  // Let us now execute the Host program. When multidevice is requested, we
+  // need to pass a pointer to a Communicator
+  HostIrExecutor hie(std::move(hic), communicator_);
+  auto outputs =
+      hie.runWithInput({{tva, aten_tva}, {tvb, aten_tvb}, {tvd, aten_tvd}});
+
+  // "validate" the result
+  EXPECT_EQ(outputs.at(0).numel(), (M * N) / communicator_->size());
+}
+
+// Let us now show how to implement Kernel Pipelining with Host IR. Let us
+// consider a program summarized as
+/*
+  | tv0: input
+  | tv1 = Fusion0 (tv0)
+  | tv2 = Fusion1 (tv1)
+  | tv2: output
+*/
+// and assume it is pointwise on the outermost dimension (i.e. tv0, tv1 and tv2
+// share the "same" outermost axis) What we want to achieve is to tile the
+// tensor and to iterate over the tiled dimension with a host for-loop, changing
+// stream at each iteration:
+/*
+  | tv0: input
+  | tv2: pre-allocated output
+  | For (int i=0; i < tv0->axis(0)->extent; i++):
+  |   Change Stream
+    // here [i, ...] means we select index i on the outermost dim
+  |   tv1_i = Fusion0 (tv0[i,...])
+  |   tv2[i,...] = Fusion1 (tv1_i)
+*/
+// To do so, we will be using new Host IRs: Stream (a Val), SetStream, ForLoop.
+TEST_F(MultiDeviceTutorial, HostIrKernekPipelining) {
+  constexpr int64_t kNDims = 2;
+  constexpr int64_t kPipelineAxis = 0;
+  constexpr int64_t kNumberOfStreams = 4;
+
+  auto CreateFusion0 = []() -> std::unique_ptr<Fusion> {
+    auto fusion = std::make_unique<Fusion>();
+    FusionGuard fg(fusion.get());
+    TensorView* tv0 = makeSymbolicTensor(
+        kNDims - 1); // kNDims-1 because the fusion receives a "selected" tensor
+    fusion->addInput(tv0);
+    auto tv1 = add(tv0, IrBuilder::create<Val>(1.));
+    fusion->addOutput(tv1);
+    return fusion;
+  };
+  auto CreateFusion1 = []() -> std::unique_ptr<Fusion> {
+    auto fusion = std::make_unique<Fusion>();
+    FusionGuard fg(fusion.get());
+    TensorView* tv1 = makeSymbolicTensor(
+        kNDims - 1); // kNDims-1 because the fusion receives a "selected" tensor
+    TensorView* tv2 = makeSymbolicTensor(kNDims - 1);
+    fusion->addInput(tv1);
+    fusion->addInput(tv2);
+    auto tv2_alias = add(tv1, IrBuilder::create<Val>(2.));
+    fusion->addOutput(tv2_alias);
+    fusion->aliasOutputToInput(tv2_alias, tv2, AllocationType::ReuseBuffer);
+    return fusion;
+  };
+
+  // Instantiate an HostIrContainer
+  auto hic = std::make_unique<HostIrContainer>();
+  FusionGuard fg(hic.get());
+
+  TensorView* tv0 = makeSymbolicTensor(kNDims);
+  TensorView* tv2 = makeSymbolicTensor(kNDims);
+
+  // Let us create the main for-loop of the program. Its index will be used in
+  // the for-loop's body.
+  auto* index = IrBuilder::create<Val>(DataType::Index);
+  auto* for_loop = IrBuilder::create<ForLoop>(
+      /*IterDomain=*/tv2->axis(0),
```

Comment:
```suggestion
      tv2->axis(0),
```
I'm surprised https://clang.llvm.org/extra/clang-tidy/checks/bugprone/argument-comment.html didn't kick in because the argument name is `iter_domain`. 

In this case, the constructor takes one `IterDomain*` so this comment is not necessary.